### ISSUE TITLE
Warning/Error handling in templates

### ISSUE THREAD

**Issue by erikbosch** [COVESA/ifex#29]
I assume the general approach for our generators is that the tooling first will be checked to assure that it is correct VSC. But then there might be limitations that are template-specific, e.g. that a certain VSC construct cannot be represented in target language X.

Issuing warnings/errors from within Jinja templates is not trivial, and it would likely be good to have a standardized approach on how templates shall reports warnings/errors/exceptions, so that they behave in the same way.

**Comment by gunnarx** [comment]
With the latest implementation in #32 we have slightly less error checking so we can't get messages like [this](https://github.com/COVESA/vsc-tools/blob/master/model/vsc_parser.py#L272) or [this](https://github.com/COVESA/vsc-tools/blob/dfc9f4e556f5f4f47f5c7422d3ca1496d21deb1b/vsc/model/vsc_ast.py#L186)

We might discuss if we need to improve this again but for now we have to work from #32 as a new starting point because so many other things have changed (improved).

I have tested the setting to configure dacite library to make it at least complain about unknown keys appearing in the YAML-dict.  I will send a PR for that soon.

**Comment by gunnarx** [comment]
Of course, errors from within a faulty Jinja template are a separate thing, but also this we should look into.  For now, referencing a variable that does not exist in the data class should will of course cause an exception, and calling gen() function with an 'unknown' type also causes an error.  I think the trade-off is that it might be OK that writing a Jinja template is hard, since that is done less often, but for user-input (YAML files) it is more important to be user friendly. IMHO.

**Comment by gunnar-mb** [comment]
- No more input received on this.
- There is a slow "trend" towards model-to-model transformation implementations now, as opposed to Jinja template based.
I believe the Jinja (text-template) approach will remain only for simpler cases which may therefore not require more advanced error-handling in the Jinja generation side.
Therefore - closing issue.
